ONSdigital/eq-author-app
# EAR-2130-implement-prepop-schema-processing

### What is the context of this PR?

This PR is linked to this [Jira ticket](https://jira.ons.gov.uk/browse/EAR-2130)

### How to review

Checkout `EAR-2130-implement-prepop-schema-processing`

- Enable the `dataset` feature flag and:
  - navigate to the ONS dataset page
  - from the survey id drop drown select a survey and link one of the datasets
  - check that the data fields text and table is displayed
- Open the author schema while the dataset is still linked and:
  - ensure that the prepop schema has the following fields id, surveyId, version, datecreated and data
  - under the `data` block, ensure that the array is populated with objects that have one of `company_name` and `company_type` as the `fieldName`
     - Check that the `exampleValue` and `fieldName` match the values displayed on the front end 
     - ensure that the `company_name` and `company_type` objects adopt their respective attributes which can be found in the `properties` block in this [SDS schema](https://sds-api-mock-gateway-74jbrn8r.nw.gateway.dev/schemaVersionGet?id=121) 
- Confirm that when you unlink the dataset, the `prepopSchema` block is deleted

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
